cosmwasm/contracts/wormchain-accounting/src/contract.rs

```diff
@@ -117,20 +117,28 @@ fn submit_observations(
     let observations: Vec<Observation> =
         from_binary(&observations).context("failed to parse `Observations`")?;
 
-    let events = observations
-        .into_iter()
-        .map(|o| {
-            let key = transfer::Key::new(o.emitter_chain, o.emitter_address.into(), o.sequence);
+    let mut details = Vec::with_capacity(observations.len());
+    let mut events = Vec::with_capacity(observations.len());
+    for o in observations {
+        let key = transfer::Key::new(o.emitter_chain, o.emitter_address.into(), o.sequence);
+        let (status, event) =
             handle_observation(deps.branch(), o, guardian_set_index, quorum, signature)
-                .with_context(|| format!("failed to handle observation for key {key}"))
-        })
-        .filter_map(Result::transpose)
-        .collect::<anyhow::Result<Vec<_>>>()
-        .context("failed to handle `Observation`")?;
+                .with_context(|| format!("failed to handle observation for key {key}"))?;
+        details.push(TransferDetails {
+            key,
+            status: Some(status),
+        });
+        if let Some(evt) = event {
+            events.push(evt);
+        }
+    }
+
+    let data = to_binary(&details).context("failed to serialize transfer details")?;
 
     Ok(Response::new()
         .add_attribute("action", "submit_observations")
         .add_attribute("owner", info.sender)
+        .set_data(data)
```

Comment:
TransferDetails can be a pretty large object. Have we considered the impact of this on the transaction result store of validators / gas costs in the wasm environment?

Reply:
Related to this, is there a suggested limit to the size of a batch that the guardian should submit?

Reply:
@hendrikhofstadt I did consider this when making this change.  I think we can make the response much smaller by not returning any data with the `TransferStatus::Pending` variant.  Then the only data would be the actual transfer details + digest for committed transfers. It's not clear if the on-chain state for pending transfers is that useful outside of testing anyway.

@bruce-riley The contract itself has no limits.  We're only limited by the max transaction size and the gas limit for the tx.

Reply:
Actually while thinking about this some more I found a couple of places where we can reduce the size of the on-chain state, which would also reduce the amount of data returned in a TransferStatus (see [1](https://github.com/wormhole-foundation/wormhole/pull/2241/commits/8b442fc3c4429c1baeebc8cabf8969cd84242ecb) and [2](https://github.com/wormhole-foundation/wormhole/pull/2241/commits/80e29d835dbcfef8828973d7c832ad887194f6c0)). 

We can still drop the data returned in the `TransferStatus::Pending` variant if we feel like it's still too much data in the response.

Reply:
I don't think the guardian needs any additional info with the pending ones. Actually, I'm not even sure it needs anything with the completed ones (A digest mismatch would throw an error, right?).

Reply:
Yes, a digest mismatch would throw an error in the completed case.

Basically it just boils down to whether we want to re-use the types from the queries or whether we want a new type that's basically just `TransferStatus` but without the data.

Reply:
I've changed the enum here to `ObservationStatus` where the `Pending` and `Committed` variants don't have any data associated with them.  It also includes an `Error` variant with the error string.

This should significantly reduce the amount of data returned by the contract and also let us keep processing the rest of the batch if an observation fails.  Not failing on error also means we can now emit an event that guardians can listen for when an observation fails.